database/model.js

```diff
@@ -10,31 +10,58 @@ const carouselSchema = new mongoose.Schema({
 
 const Carousels = mongoose.model( 'Carousel', carouselSchema );
 
-
 // findAll retrieves all appimages data
-function findAll(callback) {
-  Carousels.find({}, callback);
-}
+function findAll() {
+  return Carousels.find({})
+    .catch((err) => {
+      console.log('error inside model.findAll(): ', err);
+    });
+};
 
 // findOne will retrieve the appimage associated with the given id
-function findOne(id, callback) {
-  Carousels.find({ id: id }, callback);
-}
-// db.find({ id: id })
+const findOne = async (id, callback) => {
+  return Carousels.find({ id })
+    .catch((err) => {
+      console.log('error inside model.findOne(): ', err);
+      return err;
+    });
+};
 
 // insertOne inserts one appImages schema into db
-function insertOne(schema, callback) {
-  Carousels.create(schema, callback);
-}
+const insertOne = async (data) => {
+  return Carousels.create(data)
+    .catch((err) => {
+      console.log('error inside model.insertOne(): ', err);
+      return err;
+    });
+};
+
+const updateCarousel = async (id, data) => {
+  return Carousels.updateOne({ id }, data)
+    .catch((err) => {
+      console.log('error inside model.updateCarousel(): ', err);
+      return err;
+    });
+};
+
+const deleteCarousel = async (id) => {
+  return Carousels.deleteOne({ id })
+    .catch((err) => {
+      console.log('error inside model.deleteCarousel(): ', err);
+    });
+};
```

Comment:
I really like how you refactored the existing functions to be promise based, it looks much cleaner. 

I'm a bit confused about using 'async' without an 'await' inside of the function. It looks like you're returning a promise from the function, rather than 'awaiting' it's completion?

Reply:
Yeah I've gotten in the habit of using the 'async' keyword with my functions that I know I want to return a promise, because it guarantees that they return a promise, and it kind of reminds me when I'm calling those functions that a promise is going to be returned.. Things like doing MongoDB queries are what are creating the promises, so there isn't anywhere in the code that explicitly says "return new Promise()" etc.. the promise is created automatically so it's nice to have a reminder that that's what I'm working with any time I'm looking at that function. Also, if later you want to use await in any way inside the function, it's already set up to handle it.